src/packet_analysis/Manager.h

```diff
@@ -0,0 +1,176 @@
+// See the file "COPYING" in the main distribution directory for copyright.
+
+#pragma once
+
+#include <queue>
+#include <vector>
+
+#include "Config.h"
+#include "Tag.h"
+#include "Component.h"
+#include "plugin/ComponentManager.h"
+#include "iosource/Packet.h"
+#include "Dict.h"
+#include "net_util.h"
+
+namespace zeek::packet_analysis {
+
+class Analyzer;
+class Dispatcher;
+
+class Manager : public plugin::ComponentManager<Tag, Component> {
+public:
+	/**
+	 * Constructor.
+	 */
+	Manager();
+
+	/**
+	 * Destructor.
+	 */
+	~Manager();
+
+	/**
+	 * Second-stage initialization of the manager. This is called late
+	 * during Zeek's initialization after any scripts are processed.
+	 */
+	void InitPostScript();
+
+	/**
+	 * Finished the manager's operations.
+	 */
+	void Done();
+
+	/**
+	 * Dumps out the state of all registered analyzers to the \c analyzer
+	 * debug stream. Should be called only after any \c zeek_init events
+	 * have executed to ensure that any of their changes are applied.
+	 */
+	void DumpDebug(); // Called after zeek_init() events.
+
+	/**
+	 * Enables an analyzer type. Only enabled analyzers will be
+	 * instantiated for new connections.
+	 *
+	 * @param tag The analyzer's tag.
+	 *
+	 * @return True if successful.
+	 */
+	bool EnableAnalyzer(const Tag& tag);
+
+	/**
+	 * Enables an analyzer type. Only enabled analyzers will be
+	 * instantiated for new connections.
+	 *
+	 * @param tag The analyzer's tag as an enum of script type \c
+	 * Analyzer::Tag.
+	 *
+	 * @return True if successful.
+	 */
+	bool EnableAnalyzer(EnumVal* tag);
```

Comment:
Do the Enable/Disable/IsEnabled methods only exist to keep the interface similar to the existing L3 analyzers? They're not used anywhere that I can find. Is this part of the discussion about analyzers not being dynamically instantiated?

Reply:
Yes to both questions. I think dynamically instantiating packet analyzers won't work for performance reasons. Thus, I would say we can just remove them.